src/Common/ThreadPool.cpp

```diff
@@ -397,7 +397,10 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
 
             /// We don't run jobs after `shutdown` is set, but we have to properly dequeue all jobs and finish them.
             if (shutdown)
+            {
+                job_is_done = true;
                 continue;
+            }
```

Comment:
Without `job_is_done` `scheduled_jobs` won't be decremented and `job_finished.notify_all()` won't be called, so `ThreadPool::wait` may hang.